Develop2

BlackSunVRN opened the pull request.

BlackSunVRN committed:
Added BaseApiClient, UserClient, OrderClient
Added models for Users and Orders
Added tests for:
* create/update/login/remove users
* create/get orders

BlackSunVRN committed:
add allure report

BlackSunVRN committed:
Deleted .idea folder

pomegranate282922 commented on src/main/java/ru/yandex/practicum/stellaburgers/api/client/OrderClient.java:
```diff
@@ -0,0 +1,59 @@
+                .spec(getReqSpecJSON())
+                .body(order)
+                .when()
+                .post("/api/orders");
```
Урлы и их части вынеси в отдельный класс - так намного удобнее вносить изменения при необходимости.

BlackSunVRN replied:
Перенес все урлы и их части в BaseApiClient. Те, что используются в рамках пакета сделал protected.

pomegranate282922 commented on src/main/java/ru/yandex/practicum/stellaburgers/api/model/User.java:
```diff
@@ -0,0 +1,53 @@
+
+    public void setName(String name) {
+        this.name = name;
+    }
```
Можно использовать библиотеку lombok чтобы уменьшить количество boilerplate кода

BlackSunVRN replied:
Подключил lombock в проект. Использовал констукторы и геттеры/сеттеры в пакете model

pomegranate282922 commented on src/test/java/ru/yandex/practicum/UserDeleteTest.java:
```diff
@@ -0,0 +1,49 @@
+        Response responseDelete = userClient.deleteUser(user);
+        assertEquals(SC_ACCEPTED, responseDelete.statusCode());
+
+        try {
```
Не очень понятно, что ты хочешь этим кодом проверить. В любом случае, тест останется зелёным, а логи никто особо читать не будет)

BlackSunVRN replied:
Хотел после удаления проверить получение данных по пользователю.
Переделал на попытку логина под данными того пользователя, которого удалили.
В случае, если пользователь не был удален, мы не получим 401 и тест упадет.
В самом удалении также проверяется, что ответ на запрос был 202.
То есть, я считаю, что с проверкой кода ответа на удаление и логин точно станет понятно прошел ли тест.

pomegranate282922 commented on src/test/java/ru/yandex/practicum/UserErrorsTest.java:
```diff
@@ -0,0 +1,51 @@
+    @DisplayName("Создание пользователя без имени")
+    public void userCreateWithoutNameTest() {
+        Response responseCreate = userClient.createUserManual(user.getEmail(), user.getPassword());
+        assertEquals(SC_FORBIDDEN, responseCreate.statusCode());
```
Да, а если поймается баг и юзер создастся? Его тоже нужно удалить.

BlackSunVRN replied:
Сделал как и в 3 задании, вынес в @After удаление юзера и проверку на IllegalArgumentException

pomegranate282922 commented on src/test/java/ru/yandex/practicum/UserUpdateTest.java:
```diff
@@ -0,0 +1,164 @@
+
+        SuccessResponse updateUserResponse = responseUpdate.as(SuccessResponse.class);
+        assertFalse(updateUserResponse.success);
+
```
Delete просится в after

BlackSunVRN replied:
Из-за изменений почты и пароля в тестах просто стандартный deleteUser(user) вынести в @After было сложно, поэтому в конце каждого теста удалялись пользователи с уже измененными данными.
Сделал, как ты попросила поправить, но для этого в конце каждого теста добавились строки с кодом, в котором имеющемуся user-у передаю новые значения почты/пароля. Теперь все удаляется в @After, но мне кажется со стороны уменьшения строк кода не совсем выгодно получилось.

pomegranate282922 reviewed (commented):
Отлично! Пара замечаний набралась - поправишь?

BlackSunVRN committed:
add lombok to pom.xml and wrote @Getter/@Setter and @Constructor to all models
move all url and its parts to BaseApiClient for comfortable changes
change delete user test for check login after delete
add delete user after creation error at UserErrorsTest
move delete user to @After in UserUpdateTest

pomegranate282922 commented on src/main/java/ru/yandex/practicum/stellaburgers/api/model/SuccessResponse.java:
```diff
@@ -0,0 +1,9 @@
+@AllArgsConstructor
+public class SuccessResponse {
+
+    public boolean success;
```
Все же согласно принципу инкапсуляции поля делают как правило приватными.

pomegranate282922 reviewed (approved):
Зачёт!
